### PR TITLE
tts: unstable-2020-06-17 -> 0.0.9

### PR DESCRIPTION
**Pull request by Mic92**
~We need to wait until a function model is released: https://github.com/mozilla/TTS/pull/592#issuecomment-766327178~

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
umap-learn is also updated here: https://github.com/NixOS/nixpkgs/pull/110688

File: pkgs/tools/audio/tts/default.nix [hunk 1]
```diff
@@ -47,35 +47,30 @@ python3Packages.buildPythonApplication rec {
...
     owner = "mozilla";
     repo = "TTS";
-    rev = "72a6ac54c8cfaa407fc64b660248c6a788bdd381";
-    sha256 = "1wvs264if9n5xzwi7ryxvwj1j513szp6sfj6n587xk1fphi0921f";
+    rev = "d320ccf6c66671087ee69561b651f7f877a584af";
+    sha256 = "sha256-11WajttY2Y0Ng4vOnFX11A3VnMbd3BIDSxt24ScdOaM=";
   };
 
-  patches = [
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/tools/audio/tts/default.nix:60]: Can fetchpatch be removed from inputs?

File: pkgs/tools/audio/tts/default.nix [hunk 2]
```diff
@@ -88,17 +83,23 @@ python3Packages.buildPythonApplication rec {
...
+    pyworld
   ];
 
   postInstall = ''
     cp -r TTS/server/templates/ $out/${python3.sitePackages}/TTS/server
+    # cython modules are not installed for some reasons
+    (
+      cd TTS/tts/layers/glow_tts/monotonic_align
+      ${python3Packages.python.interpreter} setup.py install --prefix=$out
+    )
```
Comment (SuperSandro2000) [inline pkgs/tools/audio/tts/default.nix:102]: I am not sure if the subshell could break things. Does it forward everything correctly?

File: pkgs/tools/audio/tts/default.nix [hunk 3]
```diff
@@ -114,6 +115,18 @@ python3Packages.buildPythonApplication rec {
     "test_parametrized_gan_dataset"
   ];
 
+  preCheck = ''
+    # use the installed TTS in $PYTHONPATH instead of the one from source to also have cython modules.
+    rm -rf TTS
```
Comment (SuperSandro2000) [inline pkgs/tools/audio/tts/default.nix:120]:
```suggestion
    rm -r TTS
```

We want to notice when it is gone.

File: pkgs/tools/audio/tts/default.nix (hunk 2 above)
Comment (Mic92) [reply pkgs/tools/audio/tts/default.nix:102]: Yes: 

```
$ bash -c '(false)'; echo $?
1
```

File: pkgs/development/python-modules/pyworld/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  };
+
+  nativeBuildInputs = [
+    cython
+  ];
+
+  propagatedBuildInputs = [
+    numpy
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyworld/default.nix:24]:
```suggestion

  pythonImportsCheck = [ "pyworld" ];

```

nice to have to make sure it imports.

**Pull request merged**